### DISCUSSION TITLE
What are the plans for plotting going forward?

### DISCUSSION THREAD

**Discussion by lucas-wilkins** [SasView/sasview#2244] [Code Camp Topics]
What are the future plans for plotting?

I have been thinking about using VisPy for openGL things, but it will be worth considering the other options so that we have uniformity across functionality.

**Comment by butlerpd** [comment]
Other options that have been discussed that I remember are SILEX (from ESRF) and plotly? I think Robert had yet another package that he had implemented that would significantly speed up the 3D graphics in the scattering calculator which is currently dead slow, but then dropped because of this very question: is that the package to settle on for the project. Maybe somebody more familiar with what he was doing and why can add that to the list?

**Comment by krzywon** [comment]
This might not be an issue if we redo the plotting, but I wanted to mention it here. As alluded to in #2266, currently, when plots are redrawn, the `plot()` method is called, which is the same method called on plot creation. This resets a number of plot characteristics, forcing us to capture many different plot attributes (legend size, location, visibility, plot ranges, etc.) in order for plots to look the same between redraws.

Options I can think of, with my suggestion in bold:
- Keep the current scheme, but pass existing plots on redraw, taking the settings from the original plot.
- **Set the base plot attributes in `__init__()`.**
- Add a 'redraw' method in the Plotter class that only changes what is needed.
- Finer-grained class methods specific for each plot feature change.

**Comment by llimeht** [comment]
@krzywon Interesting - calling `plot()` repeatedly shouldn't necessary in lots of cases, and also carries some nasty performance and memory issues. Where is it doing this?
